semantic_conventions.md

```diff
@@ -0,0 +1,82 @@
+
+### Versioning
+
+Changes to this file affect the OpenTracing specification version. Additions should bump the minor version, and backwards-incompatible changes (or perhaps very large additions) should bump the major version.
+
+## Standard Span tags and log fields
+
+### Span tags table
+
+Span tags apply to **the entire Span**; as such, they apply to the entire timerange of the Span, not a particular moment with a particular timestamp: those sorts of events are best modelled as Span log fields (per the table in the next subsection of this document).
+
+| Span tag name | Type | Notes and examples |
+|:--------------|:-----|:-------------------|
+| `component` | string  | The software package, framework, library, or module that generated the associated Span. E.g., `"grpc"`, `"django"`, `"JDBI"`. |
+| `db.instance` | string | Database instance name. E.g., In java, if the jdbc.url="jdbc:mysql://127.0.0.1:3306/customers", the instance name is "customers". |
+| `db.statement` | string | A database statement for the given database type. E.g., for db.type="SQL", "SELECT * FROM wuser_table"; for db.type="redis", "SET mykey 'WuValue'". |
+| `db.type` | string | Database type. For any SQL database, "sql". For others, the lower-case database category, e.g. "cassandra", "hbase", or "redis". |
+| `db.user` | string | Username for accessing database. E.g., "readonly_user" or "reporting_user" |
+| `error` | bool | "true" if and only if the associated Span is in an error state |
+| `http.method` | string | HTTP method of the request for the associated Span. E.g., `"GET"`, `"POST"` |
+| `http.status_code` | integer | HTTP response status code for the associated Span. E.g., 200, 503, 404 |
+| `http.url` | string | URL of the request being handled in this segment of the trace, in standard URI format. E.g., '"https://domain.net/path/to?resource=here"' |
+| `peer.hostname` | string | Remote hostname. E.g., "opentracing.io", "internal.dns.name" |
+| `peer.ipv4` | string | Remote IPv4 address as a .-separated tuple. E.g., "127.0.0.1" |
+| `peer.ipv6` | string | Remote IPv6 address as a string of colon-separated 4-char hex tuples. E.g., "2001:0db8:85a3:0000:0000:8a2e:0370:7334" |
+| `peer.port` | integer | Remote port. E.g., 80 |
+| `peer.service` | string | Remote service name (for some unspecified definition of "service"). E.g., "elasticsearch", "a_custom_microservice", "memcache" |
+| `sampling.priority` | integer | If greater than 0, a hint to the Tracer to do its best to capture the trace. If 0, a hint to the trace to not-capture the trace. If absent, the Tracer should use its default sampling mechanism. |
+| `span.kind` | string | Either "client" or "server" for the appropriate roles in an RPC. E.g., "client", "server" |
```

Comment:
My thought was that, if/when the generated code actually happens, it would have a big block comment pointing to the `.md` file for semantic details, esp about combinations of tags/fields.